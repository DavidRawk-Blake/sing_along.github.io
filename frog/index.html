<!doctype html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Frog Goes Ribbit - Sing Along</title>
  <link rel="stylesheet" href="style.css">
  <script src="lyrics-data.js"></script>
  <script src="../js/lyrics-engine.js"></script>
  <script src="../js/karaoke-controller.js"></script>
</head>

<body>
  <!-- Back Button -->
  <button id="backBtn" class="control-button back-button" onclick="window.location.href='../index.html'"
    title="Back to Games">‹‹</button>

  <!-- Timestamp Display for Debugging -->
  <div id="timestampDisplay" class="timestamp-display">0.00s</div>

  <!-- Recognition Counter Display -->
  <div id="recogniseCounter" class="recognise-counter">Recognize: 0 words</div>

  <div class="outside-notes">
    <div class="floating-fly" style="transform: rotate(45deg);">🪰</div>
    <div class="floating-fly">🪰</div>
    <div class="floating-fly" style="transform: rotate(-30deg);">🪰</div>
    <div class="floating-fly" style="transform: rotate(90deg);">🪰</div>
    <div class="floating-fly">🪰</div>
    <div class="floating-fly" style="transform: rotate(-60deg);">🪰</div>
    <div class="floating-fly" style="transform: rotate(120deg);">🪰</div>
    <div class="floating-fly">🪰</div>
  </div>

  <div class="kids-frame-container">
    <div class="wavy-purple-frame">
      <div class="wavy-border"></div>
      <div class="pond">

      </div>

      <div class="frame-content">
        <h1>🐸 Frog Goes Ribbit - Lets Sing Along 🐸</h1>

        <div class="fun-decorations">
          <div class="musical-note">🎵</div>
          <div class="musical-note">🎶</div>
          <div class="musical-note">♪</div>
          <div class="musical-note">♫</div>
          <div class="hearts">💚</div>
          <div class="hearts">💙</div>
        </div>
        <p style="font-size: 0.9em; transform: rotate(-1deg); height: 100pt;">This is where we could put some dancing animation. 🎨</p>

        <!-- Lyrics Display Area -->
        <div class="lyrics-container">
          <div id="sentenceDisplay" class="sentence-display">
            Click "Play" to begin your sing-along experience! 🎤
          </div>
          <div class="progress-container">
            <div id="progressFill" class="progress-fill"></div>
          </div>
        </div>

        <div class="media-controls">
          <button id="playBtn" class="control-button play-button" onclick="togglePlayPause()" title="Play/Pause">▶</button>
          <button id="restartBtn" class="control-button restart-button" onclick="restartAction()"
            title="Previous/Reset">◀</button>
        </div>
      </div>
    </div>
  </div>

  <!-- Audio elements (hidden) -->
  <div style="display: none;">
    <audio id="song" preload="auto">
      Your browser does not support the audio element.
    </audio>
    <audio id="music" preload="auto">
      Your browser does not support the audio element.
    </audio>
  </div>
</body>
</html>