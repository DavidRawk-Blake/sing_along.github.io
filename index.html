<!doctype html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Sing Along</title>
  <link rel="stylesheet" href="style.css">
</head>

<body>
  <!-- Timestamp Display for Debugging -->
  <div id="timestampDisplay" class="timestamp-display">0.00s</div>
  
  <div class="outside-notes">
    <div class="floating-note">🎵</div>
    <div class="floating-note">🎶</div>
    <div class="floating-note">♪</div>
    <div class="floating-note">♫</div>
    <div class="floating-note">🎼</div>
    <div class="floating-note">♬</div>
    <div class="floating-note">🎤</div>
    <div class="floating-note">🎧</div>
  </div>

  <div class="kids-frame-container">
    <div class="wavy-purple-frame">
      <div class="wavy-border"></div>
      <div class="grass-field">
        
      </div>

      <div class="frame-content">
        <h1>🎼 Lets Sing Along 🎼</h1>

        <div class="fun-decorations">
          <div class="musical-note">🎵</div>
          <div class="musical-note">🎶</div>
          <div class="musical-note">♪</div>
          <div class="musical-note">♫</div>
          <div class="hearts">💜</div>
          <div class="hearts">💖</div>
        </div>
        <p>This is a super cool wavy purple picture frame that looks like it was made just for kids! The wiggly borders
          and fun colors make everything inside look extra special! 🎵🎨</p>
        <p style="font-size: 0.9em; transform: rotate(-1deg);">
          Put your favorite drawings, photos, or messages in here! 🖍️📸🎶
        </p>

        <!-- Lyrics Display Area -->
        <div class="lyrics-container">
          <div id="sentenceDisplay" class="sentence-display">
            Click "Play" to begin your sing-along experience! 🎤
          </div>
          <div class="progress-container">
            <div id="progressFill" class="progress-fill"></div>
          </div>
        </div>

        <div class="media-controls">
          <button id="playBtn" class="control-button play-button" onclick="playAction()" title="Play">▶</button>
          <button id="pauseBtn" class="control-button pause-button" onclick="pauseAction()" title="Pause">⏸</button>
          <button id="restartBtn" class="control-button restart-button" onclick="restartAction()" title="Rewind 5 seconds (or press ← key)">⏪</button>
        </div>
      </div>
    </div>
  </div>

  <!-- Audio elements with native controls (hidden) -->
  <div class="audio-section" style="display: none;">
    <h3>🎵 Audio Tracks</h3>
    <div class="audio-player">
      <label for="song1">Song 1:</label>
      <audio id="song1" controls preload="auto">
        <source src="song1.mp3" type="audio/mpeg">
        Your browser does not support the audio element.
      </audio>
    </div>
    <div class="audio-player">
      <label for="song2">Song 2:</label>
      <audio id="song2" controls preload="auto">
        <source src="song2.mp3" type="audio/mpeg">
        Your browser does not support the audio element.
      </audio>
    </div>
  </div>


</body>

<script src="lyrics-engine.js"></script>
<script>
  // Audio control functionality for the musical picture frame
  // Global variables for audio elements and lyrics
  let song1, song2;
  let lyricsEngine;

  // Initialize when DOM is loaded
  document.addEventListener('DOMContentLoaded', async function () {
    // Get references to audio elements
    song1 = document.getElementById('song1');
    song2 = document.getElementById('song2');

    // Initialize lyrics engine
    lyricsEngine = new LyricsEngine();
    
    // Load lyrics data first
    const dataLoaded = await lyricsEngine.loadLyricsData();
    if (!dataLoaded) {
      console.warn('Failed to load lyrics data, using fallback');
      // Show error message in sentence display
      document.getElementById('sentenceDisplay').innerHTML = '⚠️ Failed to load lyrics data. Please refresh the page.';
    } else {
      // Set audio sources dynamically from loaded data
      if (lyricsEngine.lyricsData.song_1_source) {
        song1.src = lyricsEngine.lyricsData.song_1_source;
        console.log(`Song 1 source set to: ${song1.src}`);
      }
      if (lyricsEngine.lyricsData.song_2_source) {
        song2.src = lyricsEngine.lyricsData.song_2_source;
        console.log(`Song 2 source set to: ${song2.src}`);
      }
      
      // Verify audio sources are accessible
      const audioReady = await lyricsEngine.verifyAudioSources();
      if (!audioReady) {
        console.warn('Some audio files may not be accessible - karaoke may not work properly');
      }
    }
    
    // Initialize with DOM elements after data is loaded
    lyricsEngine.initialize({
      sentenceDisplay: document.getElementById('sentenceDisplay'),
      progressFill: document.getElementById('progressFill'),
      audioPlayer: song1, // Use song1 as primary for timing
      timestampDisplay: document.getElementById('timestampDisplay')
    });

    // Reset when songs end
    song1.addEventListener('ended', () => {
      song1.currentTime = 0;
      console.log('Song1 ended and reset to beginning');
    });

    song2.addEventListener('ended', () => {
      song2.currentTime = 0;
      console.log('Song2 ended and reset to beginning');
    });

    // Add pressed effect to buttons on mouse/touch events
    const buttons = [document.getElementById('playBtn'), document.getElementById('pauseBtn'), document.getElementById('restartBtn')];
    buttons.forEach(button => {
      // Handle mouse events
      button.addEventListener('mousedown', () => addPressedEffect(button));
      // Handle touch events for mobile devices
      button.addEventListener('touchstart', () => addPressedEffect(button), { passive: true });
    });

    // Add keyboard support for rewind and skip functionality
    document.addEventListener('keydown', (event) => {
      if (event.key === 'ArrowLeft') {
        event.preventDefault(); // Prevent browser default behavior
        rewindFiveSeconds();
        addPressedEffect(document.getElementById('restartBtn')); // Visual feedback
        console.log('Left arrow key pressed - rewinding 5 seconds');
      } else if (event.key === 'ArrowRight') {
        event.preventDefault(); // Prevent browser default behavior
        skipForwardFiveSeconds();
        addPressedEffect(document.getElementById('playBtn')); // Visual feedback on play button
        console.log('Right arrow key pressed - skipping forward 5 seconds');
      } else if (event.code === 'Space') {
        event.preventDefault(); // Prevent page scroll
        togglePlayPause();
        console.log('Spacebar pressed - toggling play/pause');
      }
    });

    console.log('Audio controls initialized after DOM loaded');
  });



  // Start lyrics display
  function startLyrics() {
    lyricsEngine.startAnimation(() => stopLyrics());
  }

  // Pause lyrics display
  function pauseLyrics() {
    lyricsEngine.pauseAnimation();
  }

  // Stop and reset lyrics display
  function stopLyrics() {
    lyricsEngine.stopAnimation();
  }

  // Rewind function for 5-second back functionality
  function rewindFiveSeconds() {
    console.log('Rewinding 5 seconds');
    
    // Pause both songs first
    song1.pause();
    song2.pause();
    pauseLyrics();
    
    // Calculate new time (minimum 0 seconds)
    const newTime = Math.max(0, song1.currentTime - 5);
    
    // Set both audio tracks to new time
    song1.currentTime = newTime;
    song2.currentTime = newTime;
    
    // Brief pause before resuming (smoother transition)
    setTimeout(() => {
      // Resume playback
      song1.play().catch(e => console.log('Song1 play error:', e));
      song2.play().catch(e => console.log('Song2 play error:', e));
      startLyrics();
    }, 100); // 100ms pause for smooth transition
  }

  // Skip forward function for 5-second ahead functionality
  function skipForwardFiveSeconds() {
    console.log('Skipping forward 5 seconds');
    
    // Pause both songs first
    song1.pause();
    song2.pause();
    pauseLyrics();
    
    // Calculate new time (with upper bounds checking)
    const maxTime = Math.min(song1.duration || Infinity, song2.duration || Infinity);
    const newTime = Math.min(maxTime, song1.currentTime + 5);
    
    // Set both audio tracks to new time
    song1.currentTime = newTime;
    song2.currentTime = newTime;
    
    // Brief pause before resuming (smoother transition)
    setTimeout(() => {
      // Resume playback
      song1.play().catch(e => console.log('Song1 play error:', e));
      song2.play().catch(e => console.log('Song2 play error:', e));
      startLyrics();
    }, 100); // 100ms pause for smooth transition
  }

  // Global functions for onclick handlers
  function playAction() {
    console.log('Playing both songs simultaneously with lyrics');
    song1.play().catch(e => console.log('Song1 play error:', e));
    song2.play().catch(e => console.log('Song2 play error:', e));
    startLyrics();
    addPressedEffect(document.getElementById('playBtn'));
  }

  function pauseAction() {
    console.log('Paused both songs and lyrics');
    song1.pause();
    song2.pause();
    pauseLyrics();
    addPressedEffect(document.getElementById('pauseBtn'));
  }

  function togglePlayPause() {
    // Check if songs are currently playing by checking paused state
    // Both songs should be in sync, so checking song1 is sufficient
    if (song1.paused) {
      // Currently paused, so play
      playAction();
    } else {
      // Currently playing, so pause
      pauseAction();
    }
  }

  function restartAction() {
    rewindFiveSeconds();
    addPressedEffect(document.getElementById('restartBtn'));
  }

  // Add pressed effect functionality for better visual feedback
  function addPressedEffect(button) {
    button.classList.add('pressed');

    // Remove pressed class after short delay
    setTimeout(() => {
      button.classList.remove('pressed');
    }, 150);
  }
</script>

</html>