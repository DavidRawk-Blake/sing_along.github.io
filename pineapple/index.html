<!doctype html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Pineapple Karaoke - Sing Along</title>
  <link rel="stylesheet" href="../style/style.css">
  <link rel="stylesheet" href="style.css">
  <script src="lyrics_data.js"></script>
  <script src="../js/lyrics-engine.js"></script>
  <script src="../js/speech-recognition.js"></script>
  <script src="../js/karaoke-controller.js"></script>
</head>

<body>
  <!-- Floating tropical decorations -->
  <div class="floating-tropical">🍍</div>
  <div class="floating-tropical">🌺</div>
  <div class="floating-tropical">🥥</div>
  <div class="floating-tropical">🌴</div>
  <div class="floating-tropical">🌊</div>
  <div class="floating-tropical">☀️</div>
  <div class="floating-tropical">🏖️</div>
  <div class="floating-tropical">🐚</div>

  <!-- Back button -->
  <button class="control-button back-button" onclick="history.back()">⬅️</button>

  <!-- Debug Panel -->
  <div class="debug-panel" id="debugPanel">
    <div class="debug-toggle" onclick="toggleDebugPanel()">
      <span class="debug-summary">🔧 Debug Info</span>
    </div>
    <div class="debug-content" id="debugContent">
      <div class="debug-controls">
        <div class="timestamp-display" id="timestampDisplay">
          Current: 0.00s | Song: 0.00s
        </div>
        <button class="debug-button" onclick="pauseAudio()">⏸️</button>
        <button class="debug-button" onclick="backFiveSeconds()">⏪</button>
        <button class="debug-button" onclick="forwardFiveSeconds()">⏩</button>
      </div>
      <div class="debug-info" id="debugInfo">
        Speech Recognition Status: Not started
      </div>
      <table class="target-words-table" id="targetWordsTable">
        <thead>
          <tr>
            <th>Target Word</th>
            <th>Expected Time</th>
            <th>Recognized</th>
            <th>Matched</th>
          </tr>
        </thead>
        <tbody id="targetWordsTableBody">
          <!-- Will be populated by script -->
        </tbody>
      </table>
    </div>
  </div>



  <div class="kids-frame-container">
    <div class="wavy-tropical-frame">
      <div class="wavy-border">
        <div class="beach"></div>
        <div class="frame-content">
          <h1>🍍 Pineapple Song 🍍</h1>

          <!-- Fun decorations -->
          <div class="fun-decorations">
            <div class="floater">�</div>
            <div class="floater">🥥</div>
            <div class="floater">🌴</div>
            <div class="floater">🌊</div>
            <div class="floater">�</div>
            <div class="floater">☀️</div>
          </div>

          <!-- Lyrics container -->
          <div class="lyrics-container">
            <div class="sentence-display" id="sentenceDisplay">
              Get ready to sing along! �
            </div>
            
            <!-- Progress bar -->
            <div class="progress-container">
              <div class="progress-fill" id="progressBar"></div>
            </div>
            
            <!-- Volume indicator -->
            <div class="volume-container">
              <div class="volume-label">🎤 Your Voice Volume</div>
              <div class="volume-bar-container">
                <div class="volume-bar" id="volumeBar"></div>
              </div>
            </div>
          </div>

          <!-- Media controls -->
          <div class="media-controls">
            <button class="control-button play-button" id="playBtn">▶️</button>
            <button class="control-button restart-button" id="restartBtn">🔄</button>
          </div>
      </div>
    </div>
  </div>



  <!-- Audio element -->
  <audio id="audioPlayer" preload="auto" src="pineapple_song.mp3"></audio>

</body>

</html>