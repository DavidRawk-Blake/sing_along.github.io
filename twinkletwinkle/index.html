<!doctype html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Sing Along</title>
  <link rel="stylesheet" href="style.css">
  <script src="lyrics-data.js"></script>
  <script src="../js/lyrics-engine.js"></script>
</head>

<body>
  <!-- Back Button -->
  <button id="backBtn" class="control-button back-button" onclick="window.location.href='../index.html'" title="Back to Games">โนโน</button>
  
  <!-- Timestamp Display for Debugging -->
  <div id="timestampDisplay" class="timestamp-display">0.00s</div>
  
  <!-- Recognition Counter Display -->
  <div id="recogniseCounter" class="recognise-counter">Recognize: 0 words</div>
  
  <div class="outside-notes">
    <div class="floating-note">๐ต</div>
    <div class="floating-note">๐ถ</div>
    <div class="floating-note">โช</div>
    <div class="floating-note">โซ</div>
    <div class="floating-note">๐ผ</div>
    <div class="floating-note">โฌ</div>
    <div class="floating-note">๐ค</div>
    <div class="floating-note">๐ง</div>
  </div>

  <div class="kids-frame-container">
    <div class="wavy-purple-frame">
      <div class="wavy-border"></div>
      <div class="grass-field">
        
      </div>

      <div class="frame-content">
        <h1>๐ผ Lets Sing Along ๐ผ</h1>

        <div class="fun-decorations">
          <div class="musical-note">๐ต</div>
          <div class="musical-note">๐ถ</div>
          <div class="musical-note">โช</div>
          <div class="musical-note">โซ</div>
          <div class="hearts">๐</div>
          <div class="hearts">๐</div>
        </div>
        <p style="font-size: 0.9em; transform: rotate(-1deg); height: 200pt;">This is where we could put some dancing animation. ๐จ</p>
        
        <!-- Lyrics Display Area -->
        <div class="lyrics-container">
          <div id="sentenceDisplay" class="sentence-display">
            Click "Play" to begin your sing-along experience! ๐ค
          </div>
          <div class="progress-container">
            <div id="progressFill" class="progress-fill"></div>
          </div>
        </div>

        <div class="media-controls">
          <button id="playBtn" class="control-button play-button" onclick="playAction()" title="Play">โถ</button>
          <button id="pauseBtn" class="control-button pause-button" onclick="pauseAction()" title="Pause">โธ</button>
          <button id="restartBtn" class="control-button restart-button" onclick="restartAction()" title="Rewind 5 seconds (or press โ key)">โช</button>
        </div>
      </div>
    </div>
  </div>

  <!-- Audio elements with native controls (hidden) -->
  <div class="audio-section" style="display: none;">
    <h3>๐ต Audio Tracks</h3>
    <div class="audio-player">
      <label for="song1">Song 1:</label>
      <audio id="song1" controls preload="auto">
        Your browser does not support the audio element.
      </audio>
    </div>
    <div class="audio-player">
      <label for="song2">Song 2:</label>
      <audio id="song2" controls preload="auto">
        Your browser does not support the audio element.
      </audio>
    </div>
  </div>


</body>

<!-- Load lyrics data first (eager loading) -->
<script>
  // Audio control functionality for the musical picture frame
  // Global variables for audio elements and lyrics
  let song1, song2;
  let lyricsEngine;

  // Audio MIME type constant (all songs are MP3)
  const AUDIO_MIME_TYPE = 'audio/mpeg';

  // Function to count and update recognition words display
  function updateRecogniseCounter() {
    let recogniseCount = 0;
    
    // Count all words with recognise: true in the lyrics data
    if (window.lyricsData && window.lyricsData.sentences) {
      window.lyricsData.sentences.forEach(sentence => {
        sentence.words.forEach(word => {
          if (word.recognise === true) {
            recogniseCount++;
          }
        });
      });
    }
    
    // Update the counter display
    const counterElement = document.getElementById('recogniseCounter');
    if (counterElement) {
      counterElement.textContent = `0/${recogniseCount}`;
    }
  }

  // Initialize when DOM is loaded
  document.addEventListener('DOMContentLoaded', function () {
    // Get references to audio elements
    song1 = document.getElementById('song1');
    song2 = document.getElementById('song2');

    // Initialize lyrics engine (data is now hardcoded, so this always succeeds)
    lyricsEngine = new LyricsEngine();
    
    // Set audio sources directly from lyrics data (no loading needed)
    if (lyricsEngine.lyricsData.song_1_source) {
      song1.src = lyricsEngine.lyricsData.song_1_source;
      song1.type = AUDIO_MIME_TYPE;
      console.log(`Song 1 source set to: ${song1.src} (${AUDIO_MIME_TYPE})`);
    }
    
    if (lyricsEngine.lyricsData.music_source) {
      song2.src = lyricsEngine.lyricsData.music_source;
      song2.type = AUDIO_MIME_TYPE;
      console.log(`Song 2 source set to: ${song2.src} (${AUDIO_MIME_TYPE})`);
    }
    
    // Initialize with DOM elements after data is loaded
    lyricsEngine.initialize({
      sentenceDisplay: document.getElementById('sentenceDisplay'),
      progressFill: document.getElementById('progressFill'),
      audioPlayer: song1, // Use song1 as primary for timing
      timestampDisplay: document.getElementById('timestampDisplay')
    });

    // Update recognition counter display
    updateRecogniseCounter();

    // Reset when songs end
    song1.addEventListener('ended', () => {
      song1.currentTime = 0;
      console.log('Song1 ended and reset to beginning');
    });

    song2.addEventListener('ended', () => {
      song2.currentTime = 0;
      console.log('Song2 ended and reset to beginning');
    });

    // Add audio loading state listeners for debugging
    song1.addEventListener('loadstart', () => console.log('Loading song 1...'));
    song1.addEventListener('canplay', () => console.log('Song 1 ready to play'));
    song1.addEventListener('error', (e) => console.error('Error loading song 1:', e));
    
    song2.addEventListener('loadstart', () => console.log('Loading song 2...'));
    song2.addEventListener('canplay', () => console.log('Song 2 ready to play'));
    song2.addEventListener('error', (e) => console.error('Error loading song 2:', e));

    // Add pressed effect to buttons on mouse/touch events
    const buttons = [document.getElementById('playBtn'), document.getElementById('pauseBtn'), document.getElementById('restartBtn')];
    buttons.forEach(button => {
      // Handle mouse events
      button.addEventListener('mousedown', () => addPressedEffect(button));
      // Handle touch events for mobile devices
      button.addEventListener('touchstart', () => addPressedEffect(button), { passive: true });
    });

    // Add keyboard support for rewind and skip functionality
    document.addEventListener('keydown', (event) => {
      if (event.key === 'ArrowLeft') {
        event.preventDefault(); // Prevent browser default behavior
        rewindFiveSeconds();
        addPressedEffect(document.getElementById('restartBtn')); // Visual feedback
        console.log('Left arrow key pressed - rewinding 5 seconds');
      } else if (event.key === 'ArrowRight') {
        event.preventDefault(); // Prevent browser default behavior
        skipForwardFiveSeconds();
        addPressedEffect(document.getElementById('playBtn')); // Visual feedback on play button
        console.log('Right arrow key pressed - skipping forward 5 seconds');
      } else if (event.code === 'Space') {
        event.preventDefault(); // Prevent page scroll
        togglePlayPause();
        console.log('Spacebar pressed - toggling play/pause');
      }
    });

    console.log('Audio controls initialized after DOM loaded');
  });



  // Start lyrics display
  function startLyrics() {
    lyricsEngine.startAnimation(() => stopLyrics());
  }

  // Pause lyrics display
  function pauseLyrics() {
    lyricsEngine.pauseAnimation();
  }

  // Stop and reset lyrics display
  function stopLyrics() {
    lyricsEngine.stopAnimation();
  }

  // Rewind function for 5-second back functionality
  function rewindFiveSeconds() {
    console.log('Rewinding 5 seconds');
    
    // Pause both songs first
    song1.pause();
    song2.pause();
    pauseLyrics();
    
    // Calculate new time (minimum 0 seconds)
    const newTime = Math.max(0, song1.currentTime - 5);
    
    // Set both audio tracks to new time
    song1.currentTime = newTime;
    song2.currentTime = newTime;
    
    // Brief pause before resuming (smoother transition)
    setTimeout(() => {
      // Resume playback
      song1.play().catch(e => console.log('Song1 play error:', e));
      song2.play().catch(e => console.log('Song2 play error:', e));
      startLyrics();
    }, 100); // 100ms pause for smooth transition
  }

  // Skip forward function for 5-second ahead functionality
  function skipForwardFiveSeconds() {
    console.log('Skipping forward 5 seconds');
    
    // Pause both songs first
    song1.pause();
    song2.pause();
    pauseLyrics();
    
    // Calculate new time (with upper bounds checking)
    const maxTime = Math.min(song1.duration || Infinity, song2.duration || Infinity);
    const newTime = Math.min(maxTime, song1.currentTime + 5);
    
    // Set both audio tracks to new time
    song1.currentTime = newTime;
    song2.currentTime = newTime;
    
    // Brief pause before resuming (smoother transition)
    setTimeout(() => {
      // Resume playback
      song1.play().catch(e => console.log('Song1 play error:', e));
      song2.play().catch(e => console.log('Song2 play error:', e));
      startLyrics();
    }, 100); // 100ms pause for smooth transition
  }

  // Global functions for onclick handlers
  function playAction() {
    console.log('Playing both songs simultaneously with lyrics');
    song1.play().catch(e => console.log('Song1 play error:', e));
    song2.play().catch(e => console.log('Song2 play error:', e));
    startLyrics();
    addPressedEffect(document.getElementById('playBtn'));
  }

  function pauseAction() {
    console.log('Paused both songs and lyrics');
    song1.pause();
    song2.pause();
    pauseLyrics();
    addPressedEffect(document.getElementById('pauseBtn'));
  }

  function togglePlayPause() {
    // Check if songs are currently playing by checking paused state
    // Both songs should be in sync, so checking song1 is sufficient
    if (song1.paused) {
      // Currently paused, so play
      playAction();
    } else {
      // Currently playing, so pause
      pauseAction();
    }
  }

  function restartAction() {
    rewindFiveSeconds();
    addPressedEffect(document.getElementById('restartBtn'));
  }

  // Add pressed effect functionality for better visual feedback
  function addPressedEffect(button) {
    button.classList.add('pressed');

    // Remove pressed class after short delay
    setTimeout(() => {
      button.classList.remove('pressed');
    }, 150);
  }
</script>

</html>