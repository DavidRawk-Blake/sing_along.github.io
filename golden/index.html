<!doctype html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Golden - Sing Along</title>
  <link rel="stylesheet" href="style.css">
  <script src="lyrics-data.js"></script>
  <script src="../js/lyrics-engine.js"></script>
  <script src="../js/karaoke-controller.js"></script>
</head>

<body>
  <!-- Back Button -->
  <button id="backBtn" class="control-button back-button" onclick="window.location.href='../index.html'"
    title="Back to Games">‹‹</button>

  <!-- Debug Controls Area -->
  <div class="debug-controls">
    <div id="timestampDisplay" class="timestamp-display">0.00s</div>
    <button id="rewindBtn" class="debug-button" onclick="rewindFiveSeconds()" title="Rewind 5s">⏪</button>
    <button id="skipBtn" class="debug-button" onclick="skipForwardFiveSeconds()" title="Skip 5s">⏩</button>
    <div id="recogniseCounter" class="recognise-counter">Recognize: 0 words</div>
  </div>

  <div class="outside-notes">
    <div class="floating-note">🎉</div>
    <div class="floating-note">🎉</div>
    <div class="floating-note">💃</div>
    <div class="floating-note">🕺</div>
    <div class="floating-note">🎉</div>
    <div class="floating-note">💃</div>
    <div class="floating-note">🎉</div>
    <div class="floating-note">🕺</div>
  </div>

  <div class="kids-frame-container">
    <div class="wavy-purple-frame">
      <div class="wavy-border"></div>
      <div class="grass-field">

      </div>

      <div class="frame-content">
        <h1>🌟 Golden - Lets Sing Along 🌟</h1>

        <div class="fun-decorations">
          <div class="musical-note">🎉</div>
          <div class="musical-note">🎉</div>
          <div class="musical-note">💃</div>
          <div class="musical-note">🕺</div>
          <div class="hearts">🎉🎉</div>
          <div class="hearts">💃</div>
        </div>
        <p style="font-size: 0.9em; transform: rotate(-1deg); height: 200pt;">This is where we could put some dancing
          animation. 🎨</p>

        <!-- Lyrics Display Area -->
        <div class="lyrics-container">
          <div id="sentenceDisplay" class="sentence-display">
            Click "Play" to begin your sing-along experience! 🎤
          </div>
          <div id="progressBar" class="progress-bar">
            <div id="progressFill" class="progress-fill"></div>
          </div>
        </div>

        <div class="media-controls">
          <button id="playBtn" class="control-button play-button" onclick="togglePlayPause()" title="Play/Pause">▶</button>
          <button id="restartBtn" class="control-button restart-button" onclick="restartAction()"
            title="Previous/Reset">◀</button>
        </div>
      </div>
    </div>
  </div>

  <!-- Audio elements (hidden but required for karaoke functionality) -->
  <div style="display: none;">
    <audio id="song" preload="auto">
      <source src="song.mp3" type="audio/mpeg">
      Your browser does not support the audio element.
    </audio>
    <audio id="music" preload="auto">
      <source src="music.mp3" type="audio/mpeg">
      Your browser does not support the audio element.
    </audio>
  </div>

</body>

</html>